client/api/analytics/AnalyticsManager.cs

```diff
@@ -53,68 +51,49 @@ public void Start()
 
         public void Stop()
         {
-            if(config.analyticsEnabled && this.timer != null)
+            if (config.analyticsEnabled && this.timer != null)
             {
                 this.timer.Stop();
                 this.timer = null;
             }
         }
 
-        public void PushToQueue(dto.Target target, FeatureConfig featureConfig, Variation variation)
+        public void PushToCache(dto.Target target, FeatureConfig featureConfig, Variation variation)
         {
-            Analytics analytics = new Analytics(featureConfig, target, variation, EventType.METRICS);
-            long sequence = -1;
-            if (!ringBuffer.TryNext(out sequence)) // Grab the next sequence if we can
+            var cacheSize = analyticsCache.GetAllElements().Count;
+            var bufferSize = config.getBufferSize();
+
+            if (cacheSize > bufferSize)
             {
-                Log.Warning("Insufficient capacity in the analytics ringBuffer");
+                Log.Warning("Metric frequency map exceeded buffer size ({0} > {1}), force flushing", cacheSize, bufferSize);
+
+                // If the map is starting to grow too much then push the metrics now and reset the counters
+                SendMetrics();
```

Comment:
Looks good - just check make sure counts look right on the UI with Java I found a few weird bugs around draining the map (e.g. overcounting)